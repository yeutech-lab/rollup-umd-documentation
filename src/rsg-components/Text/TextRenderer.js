import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import cx from 'classnames';

export const styles = ({ fontFamily, fontSize, color }) => ({
	text: {
		fontFamily: fontFamily.base,
	},
	inheritSize: {
		fontSize: 'inherit',
	},
	smallSize: {
		fontSize: fontSize.small,
	},
	baseSize: {
		fontSize: fontSize.base,
	},
	textSize: {
		fontSize: fontSize.text,
	},
	baseColor: {
		color: color.base,
	},
	lightColor: {
		color: color.light,
	},
	em: {
		fontStyle: 'italic',
	},
	strong: {
		fontWeight: 'bold',
	},
	isUnderlined: {
		borderBottom: [[1, 'dotted', color.lightest]],
	},
});

export function TextRenderer({ classes, semantic, size, color, underlined, children, ...props }) {
	const Tag = semantic || 'span';
	const classNames = cx(classes.text, classes[`${size}Size`], classes[`${color}Color`], {
		[classes[semantic]]: semantic,
		[classes.isUnderlined]: underlined,
	});

	return (
		<Tag {...props} className={classNames}>
			{children}
		</Tag>
	);
}

TextRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	semantic: PropTypes.oneOf(['em', 'strong']),
	size: PropTypes.oneOf(['inherit', 'small', 'base', 'text']),
	color: PropTypes.oneOf(['base', 'light']),
	underlined: PropTypes.bool,
	children: PropTypes.node.isRequired,
};

TextRenderer.defaultProps = {
	size: 'inherit',
	color: 'base',
	underlined: false,
};

export default Styled(styles)(TextRenderer);
