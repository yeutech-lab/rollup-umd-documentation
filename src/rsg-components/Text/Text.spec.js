import React from 'react';
import { TextRenderer, styles } from './TextRenderer';

const props = {
	classes: classes(styles),
};

describe('Text', () => {
	it('should render text', () => {
		const actual = shallow(<TextRenderer {...props}>Pizza</TextRenderer>);

		expect(actual).toMatchSnapshot();
	});

	it('should render underlined text', () => {
		const actual = shallow(
			<TextRenderer {...props} underlined>
				Pizza
			</TextRenderer>
		);

		expect(actual).toMatchSnapshot();
	});

	it('should render sized text', () => {
		const actual = shallow(
			<TextRenderer {...props} size="small">
				Pizza
			</TextRenderer>
		);

		expect(actual).toMatchSnapshot();
	});

	it('should render colored text', () => {
		const actual = shallow(
			<TextRenderer {...props} color="light">
				Pizza
			</TextRenderer>
		);

		expect(actual).toMatchSnapshot();
	});

	it('should render text with a semantic tag and styles', () => {
		const actual = shallow(
			<TextRenderer {...props} semantic="strong">
				Pizza
			</TextRenderer>
		);

		expect(actual).toMatchSnapshot();
	});

	it('should render text with a title', () => {
		const actual = shallow(
			<TextRenderer {...props} title="Pasta">
				Pizza
			</TextRenderer>
		);

		expect(actual).toMatchSnapshot();
	});
});
