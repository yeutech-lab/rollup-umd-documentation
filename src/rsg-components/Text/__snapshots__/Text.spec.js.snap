// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render colored text 1`] = `
<span
  className="text inheritSize lightColor"
>
  Pizza
</span>
`;

exports[`Text should render sized text 1`] = `
<span
  className="text smallSize baseColor"
>
  Pizza
</span>
`;

exports[`Text should render text 1`] = `
<span
  className="text inheritSize baseColor"
>
  Pizza
</span>
`;

exports[`Text should render text with a semantic tag and styles 1`] = `
<strong
  className="text inheritSize baseColor strong"
>
  Pizza
</strong>
`;

exports[`Text should render text with a title 1`] = `
<span
  className="text inheritSize baseColor"
  title="Pasta"
>
  Pizza
</span>
`;

exports[`Text should render underlined text 1`] = `
<span
  className="text inheritSize baseColor isUnderlined"
>
  Pizza
</span>
`;
